packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -188,21 +215,113 @@ class FuchsiaDevice extends Device {
 
   @override
   Future<LaunchResult> startApp(
-    ApplicationPackage package, {
+    covariant FuchsiaApp package, {
     String mainPath,
     String route,
     DebuggingOptions debuggingOptions,
     Map<String, dynamic> platformArgs,
     bool prebuiltApplication = false,
     bool usesTerminalUi = true,
     bool ipv6 = false,
-  }) =>
-      Future<void>.error('unimplemented');
+  }) async {
+    if (!prebuiltApplication) {
+      await buildFuchsia(fuchsiaProject: FlutterProject.current().fuchsia,
+                         target: mainPath,
+                         buildInfo: debuggingOptions.buildInfo);
+    }
+    // Stop the app if it's currently running.
+    await stopApp(package);
+    // Find out who the device thinks we are.
+    final String host = await fuchsiaSdk.fuchsiaDevFinder.resolve(name);
+    final int port = await os.findFreePort();
+    if (port == 0) {
+      printError('Failed to find a free port');
+      return LaunchResult.failed();
+    }
+    final Directory packageRepo =
```

Comment:
If we delete + create this directory entirely in build, would it be better to use a tempdir? (don't know, asking)

Reply:
For this PR, I wanted to have it in an easy-to-find location just in case I needed to inspect the contents during development.